prov/efa/src/rdm/efa_rdm_pke_rtm.c

```diff
@@ -883,20 +881,7 @@ ssize_t efa_rdm_pke_proc_matched_mulreq_rtm(struct efa_rdm_pke *pkt_entry)
 			efa_rdm_tracepoint(runtread_read_posted, rxe->msg_id,
 				    (size_t) rxe->cq_entry.op_context, rxe->total_len);
 
-			err = efa_rdm_ope_post_remote_read_or_queue(rxe);
-			if (err) {
-				if (err == -FI_ENOMR) {
-					if (efa_rdm_peer_support_read_nack(peer))
-						/* Only set the flag here. The NACK
```

Comment:
where did runtread post read nack pkt?

Reply:
See line 919

Reply:
I see, we are only posting read nack pkt for runt read when we get all pkts that were runt. So we cannot post read nack here. This does make the refactor a bit less useful...

Reply:
It seems we don't need `efa_rdm_rxe_map_insert` for runt read either.
Emulated long-read write did not handle MR exhaustion. The refactor will check for that.

Reply:
>It seems we don't need efa_rdm_rxe_map_insert for runt read either.

We don't need `efa_rdm_rxe_map_insert` for runt read because it is a multi-req protocol and we already inserted it earlier.

> Emulated long-read write did not handle MR exhaustion. The refactor will check for that.

Yeah, I agree. I was just thinking how to make `efa_rdm_pke_post_remote_read_or_nack` cover the nack pkt post for all pkt types. It seems we cannot do that for runting read

Reply:
I am ok with documenting in `efa_rdm_pke_post_remote_read_or_nack` that why we don't need map insert and nack post for runting read.